packages/core/src/system.ts

```diff
@@ -0,0 +1,34 @@
+import { ActorSystem, ActorSystemInfo, AnyActorRef } from './types.js';
+
+export function createSystem<T extends ActorSystemInfo>(): ActorSystem<T> {
+  let sessionIdIndex = 0;
+  const children = new Map<string, AnyActorRef>();
+  const keyedActors = new Map<keyof T['actors'], AnyActorRef | undefined>();
+  const reverseKeyedActors = new WeakMap<AnyActorRef, keyof T['actors']>();
+
+  const system: ActorSystem<T> = {
+    _register: (actorRef) => {
+      const id = `x:${sessionIdIndex++}`;
+      children.set(id, actorRef);
+      return id;
+    },
+    _unregister: (actorRef) => {
+      children.delete(actorRef.id);
```

Comment:
I'm almost certain that this is a bug/mem leak. It should be:
```suggestion
      children.delete(actorRef.sessionId);
```